Add Netduino

NKnusperer opened the pull request:
Support for Netduino Plus 1 and Netduino Universal (Netduino Plus 2,
Netduino 3 Ethernet/Wi-Fi )

NKnusperer committed:
Add Netduino

Support for Netduino Plus 1 and Netduino Universal (Netduino Plus 2,
Netduino 3 Ethernet/Wi-Fi )

Golemde commented:
Wow! Ich begeistert! Vielen Dank!
Ich würde den Code gerne mergen, allerdings sind da noch zwei problematische Stellen drin:
BuildGetRequestString() setzt zwangsweise alle URL-Parameter, obwohl nur temp&dbg tatsächlich erforderlich sind. Werden die weitere Parameter gesendet, obwohl sie leer sind oder einen unbekannten Wert enthalten (token!), werden sie von temp.php verworfen. Hier sollte OfficeTemperatureConfig() defaultmäßig mit Null arbeiten und BuildGetRequestString() die URL nur die Parametern mit Nicht-Null-Werten ergänzen (Geht das so in C#?)
Außerdem sollte der city-Parameter unbedingt URL-codiert werden, ich steck zu wenig in C# drin, um zu wissen, ob das nicht irgendwo automatisch passiert.

NKnusperer committed:
Netduino support enhancements

Ignore null-values in OfficeTemperatureConfig for optional parameters
URL encode city name

NKnusperer commented:
Wird jetzt so gemacht.

Golemde commented:
Vielen Dank! Netduino ist jetzt mit drin.
